### DISCUSSION TITLE
How to make compilation error in transformer?

### DISCUSSION THREAD

**Discussion by samchon** [nonara/ts-patch#123] [Q&A] [answered]
Hello, I'm developing a transformer based library [typia](https://github.com/samchon/typia).

In nowadays, I hope to make compilation error in the `typia` side. Currently, I'm using `console.error()` function, or throw exception halting entire `tsc --watch` command. because I don't know how to make compilation error in my program side yet. If you know how, and just give me a keyword or show me an example code, it would be great helpful for me. Thanks for reading.

![image](https://github.com/nonara/ts-patch/assets/13158709/ef7e15eb-c181-47c4-af26-f12f8ae5949d)

> Hope to make a compilation error like `src/index.ts:9:7 - error TS2322: Type 'string' is not assignable to type 'number'` word.

**Comment by nonara** [comment, marked as answer]
Hello! Those errors are called Diagnostics.

Tsp gives you a helper utility called `addDiagnostic`, which should do what you need.

**Comment by samchon** [comment]
I regret that I should have read the README documentat more carefully. 

Anyway, whenever using `ts-patch`, I always thought this library has great knowhows. Even in this case, I am once again amazed at the placement and discernment of the `ts-patch` features. I just traveled TypeScript API only and have experimented features of it, so I once gave up and planned to detour this problem just by transforming to weird statement like below. 

This library is amazing. Nice to ask here.

```typescript
const TYPIA_TRANSFORMATION_ERROR: number = "typia.protobuf.encode<T>() supports only a sole and static object type"`;
```
